### ISSUE TITLE
Cannot run inside a transaction block

### ISSUE THREAD

**Issue by pauldraper** [amacneil/dbmate#46]
PostreSQL

```
-- migrate:up
ALTER TYPE my_enum ADD VALUE 'my_value';
```

Running this migrations results in

```
Error: pq: ALTER TYPE ... ADD cannot run inside a transaction block
```

**Comment by camerondavison** [comment]
is there a work around for this?

**Comment by amacneil** [comment]
For this particular issue, you could try falling back to the old way of modifying an enum, it might work inside a transaction:

https://stackoverflow.com/a/10404041

More generally, it would be nice if there was a way to disable our transaction logic (run a specific migration without automatically wrapping in a transaction). Something like:

```
-- migrate:up transaction:false
```

I'm open to suggestions on the syntax and/or a PR implementing this, it should be quite easy.

**Comment by adrianomedeirossantos** [comment]
Hi @amacneil, it would be great to have support for this feature.  I'm currently working in a project which needs to run migrations out of a transaction.  I need to run the following migration:

```
DROP INDEX CONCURRENTLY my_idx;
```

Unfortunately,  `CONCURRENTLY` cannot be executed within a transaction block as you can see in http://www.postgresqltutorial.com/postgresql-indexes/postgresql-drop-index/. 

Do you think this feature could be implemented any soon?

**Comment by pauldraper** [comment]
> -- migrate:up transaction:false

Yes I agree this is right.

**Comment by amacneil** [comment]
I think this feature would be pretty straight forward to implement - it just needs some bike shedding on the comment syntax, additional parsing logic to detect when transactions are disabled, and some light refactoring to not automatically open the transaction.

I don't currently have time to tackle it, but would happily review a PR.

**Comment by benjreinhart** [comment]
Addressed in #68
